### PR TITLE
fix(maps): correctly add new markers to the live map view

### LINKED ISSUE #7792: [map] useMemo caching is too aggressive
The result is that new markers added in another tab won't appear live in an already loaded map correctly.

I know what's causing this, I'll make a PR.

### PR DESCRIPTION
**Pull request by seagreen**
Close #7792

When the user creates a location row it appears in the DB first partially completed. The final version only appears later once the user has filled out all the columns.

Previously we were caching too early in this process. This captured partially completed objects which lacked both `latitude` and `longitude`. Thus they didn't appear on the map.

### CODE REVIEW

File: packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { useState, useEffect } from 'react';
+
+export const useStabilizedObject = <T>(obj: T, customEqual: (oldObject: T, newObject: T) => boolean) => {
```
Comment (wittjosiah) [inline packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts:7]: when making a generic util like this try to include a jsdoc about the intended use case/what it's for

File: packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  const [stableObj, setStableObj] = useState(obj);
+
+  useEffect(() => {
```
Comment (wittjosiah) [inline packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts:10]: beware that when using an effect here the returned markers above will be a render cycle behind the map and so if the map input changes there will be a render where the returned markers are for the previous map

File: packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts (hunk 1 above)
Comment (seagreen) [reply packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts:17]: Good idea, done.

File: packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts (hunk 2 above)
Comment (seagreen) [reply packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts:20]: Ooof. Added a TODO comment for now. I'd also be down for trying to get a correct implementation of `useStabilizedObject`, but maybe not in this PR unless you know exactly what to do.

**Pull request merged**